new-lamassu-admin/src/pages/AddMachine/AddMachine.js

```diff
@@ -27,12 +29,44 @@ const SAVE_CONFIG = gql`
     createPairingTotem(name: $name)
   }
 `
+const GET_MACHINES = gql`
+  {
+    machines {
+      name
+      deviceId
+    }
+  }
+`
 
 const useStyles = makeStyles(styles)
 
-const QrCodeComponent = ({ classes, qrCode, close }) => {
-  const [doneButton, setDoneButton] = useState(null)
-  setTimeout(() => setDoneButton(true), 2000)
+const getSize = R.compose(R.length, R.pathOr([], ['machines']))
+
+function usePairDevice({ name, count }) {
+  const [on, toggle] = useBoolean(false)
+  const { data, stopPolling, startPolling } = useQuery(GET_MACHINES)
+  const size = getSize(data)
+
+  useLifecycles(() => startPolling(10000), stopPolling)
+  useEffect(() =>
+    size > count && data?.machines?.some(m => m.name === name)
+      ? toggle(true)
+      : undefined
+  )
+
+  return [on]
+}
```

Comment:
I don't think we need this custom hook. 

On the QrCodeComponent you can do:
```js
const { data } = useQuery(GET_MACHINES, { pollInterval: 10000 })
if (getSize(data) > count && data?.machines?.some(m => m.name === name)) close() {
  onPaired(name)
}
```

Since polling is being handled by Apollo, it will stop on component unmount already. you can test if this is properly working by looking at the networking tab.

You can also change the `onPaired` function on the outside to actually close the popup as well, instead of manually calling for close on pair.